avalon/tools/models.py

```diff
@@ -213,19 +213,34 @@ def _get_task_icons(self):
                                       color=style.colors.default)
                 self._icons[task["name"]] = icon
 
-    def set_assets(self, asset_ids):
+    def set_assets(self, asset_ids=[], asset_entities=None):
         """Set assets to track by their database id
 
         Arguments:
             asset_ids (list): List of asset ids.
+            asset_entities (list): List of asset entities from MongoDB.
```

Comment:
The extra argument seems a bit odd. It seems explicit but also like an odd choice.

Would it make more sense to do Python duck-typing. E.g. check if it's a `dict` when provided in `asset_ids` and then assume it's an `asset_entity`. Also, should the terminology be `asset document`? I believe MongoDB refers to Documents over Entities.

Not saying it's better. I'm just wondering what others think.

Reply:
With `asset_document` I'm ok to change it and I agree it should be document.

The `asset_ids` was kept due to kind of backwards compatibility, I can imagine somebody has it's own tool using this widget. But if you don't agree, it's easy to remove whole part with `asset_ids`.